code/online_challenges/src/project_euler/problem_028/problem_028.cpp

```diff
@@ -0,0 +1,28 @@
+#include <iostream>
+#include <cmath>
+
+bool isPerfectSquare(int num)
+{
+    int sqrtNum = static_cast<int>(std::sqrt(num));
+    return sqrtNum * sqrtNum == num;
+}
+
+int main()
+{
+    int limit = 1001 * 1001;
+    int incrementRate = 0;
+    int diagonalNumberSum = 0;
+    for (int i = 1; i <= limit; i += incrementRate) // Iterate over every diagonal number
+    {
+        diagonalNumberSum += i; // Add the current diagonal number
+        if (
+            (i % 2 == 1) &&      // If the current number is odd
+            (isPerfectSquare(i)) // and it is a perfect square
```

Comment:
1. The newlines of parentheses of `if` is not needed.
2. Put logical operator in front of the conditions, Is it clearer? (If you agree, I'll add to the coding style.)

```
if ((i % 2 == 1)
    && isPerfectSquare(i))
```

Reply:
Ok, I think it should be added.